### ISSUE TITLE
Plugin causes extremely high Atom Helper CPU usage

### ISSUE THREAD

**Issue by bsiegel** [JavascriptIsMagic/tree-view-git-projects#4]
This is a retina MBP mid-2012, OSX 10.10.

Without tree-view-git-projects:
![screen shot 2015-02-09 at 3 56 48 pm](https://cloud.githubusercontent.com/assets/96068/6115568/6c7b5ba2-b074-11e4-8abb-5001f2799c89.png)

With tree-view-git-projects:
![screen shot 2015-02-09 at 3 57 22 pm](https://cloud.githubusercontent.com/assets/96068/6115570/713379b8-b074-11e4-8ce7-66a460ec9d89.png)

**Comment by JavascriptIsMagic** [comment]
as of v0.2.1 we make sure the recursive directory search loop does not pass through external functions and will always exhaust the path you feed it.

---

(Sorry it took so long for me to look at this, I am supposed to be on vacation this week, but I got antsy so I decided to do a little work today with my vacation time..)

I tracked it down, basically `path.resolve` was in some cases causing a resolve back to the current working directory, which then recursively looped back to nothing and resolved back to the correct working directory again in an infinite loop.

The package was functioning fine on my work computer when I tested it, but here on my home computer when not opening a file I experienced the same issue here. (this is particularly tricky to test due to it recursively searching file-system paths which are different on each platform as well as every difference in directory structure on each hard drive.)
